app/controllers/PropertyLinkingController.scala

```diff
@@ -105,10 +106,22 @@ class PropertyLinkingController @Inject()(
       .fold(NotFound("clients property links not found"))(propertyLinks => Ok(Json.toJson(propertyLinks)))
   }
 
-  def getClientsPropertyLink(submissionId: String): Action[AnyContent] = authenticated.async { implicit request =>
-    propertyLinkService
-      .getClientsPropertyLink(submissionId)
-      .fold(NotFound("client property link not found")) { authorisation => Ok(Json.toJson(authorisation)) }
+  def getClientsPropertyLink(submissionId: String, projection: String = "propertiesView"): Action[AnyContent] = authenticated.async { implicit request => //fixme add projection for propertyView (current behaviour) and clientProperty to support permissions
+    projection match {
+      case "propertiesView" => {
+        propertyLinkService
+          .getClientsPropertyLink(submissionId)
+          .fold(NotFound("client property link not found")) { pl => Ok(Json.toJson(PropertiesView(pl.authorisation, Nil))) }
+      }
+      case "clientsPropertyLink" => {
+        propertyLinkService
+          .getClientsPropertyLink(submissionId)
+          .fold(NotFound("client property link not found")) { pl => Ok(Json.toJson(pl.authorisation)) }
+      }
+      case _ => throw new IllegalArgumentException(s"unsupported projection: $projection")
```

Comment:
wouldn't this produce a 500 response? I think the projection should be an enum of possible / supported values and fail with 400 if the received value is not supported.

Reply:
I have updated it and I am now returning a 501 instead. It is how we have handled this in other places.